app/projectsmodel.cpp

```diff
@@ -635,3 +621,18 @@ ProjectsModel::ProjectModelTypes ProjectsModel::modelType() const
 {
   return mModelType;
 }
+
+SynchronizationManager *ProjectsModel::syncManager() const
+{
+  return mSyncManager;
+}
+
+void ProjectsModel::setSyncManager( SynchronizationManager *newSyncManager )
+{
+  if ( mSyncManager == newSyncManager )
+    return;
+  mSyncManager = newSyncManager;
+  emit syncManagerChanged( mSyncManager );
+
+  initializeProjectsModel();
```

Comment:
not sure if this call should be in setter.... Create internal connection for syncManagerChanged signal?